keps/sig-network/20190125-ingress-api-group.md

```diff
@@ -128,138 +126,127 @@ Ingress providers) treats the path match as a prefix match. For
-and PathPrefix in the API.
-
-<!--
-1) v1beta1 gets a "PathPrefix" field, specced as a strict prefix, no
-trailing "*" or ".*" needed.  Document that Path is under-specified
-and implementation-specific.  Document that all implementations are
-expected to support PathPrefix with consistent semantics.  Path and
-PathPrefix can be mutually exclusive.  v1 gets either Path or
-PathPrefix (debatable) and an annotation
-"ingress.kubernetes.io/beta-path" which maps the new field to the old.
-That satisfies round-trip requirement but is kind of horrible.
-
-2) v1beta1 gets a "PathPrefix" field as above.  v1 gets an annotation
-"ingress.kubernetes.io/beta-path-semantics".  If we create a v1 object
-from a v1beta1 where `path` was used, we set the annotation.
-Validation logic will check the annotation to decide how to interpret
-and validate paths.  We'll have to keep it for a long while.  Maybe we
-can prevent new objects from being created with that annotation?  Now
-what if implementations want to offer "extended" semantics here?  Same
-validation problem.
-
-3)  v1beta1 gets a "PathPrefix" field as above.  v1 gets an annotation
-"ingress.kubernetes.io/non-portable-path-semantics".  If we create a
-v1 object from a v1beta1 where `path` was used, we set the annotation.
-We keep it forever.
-
-4) Same as 2 but we just don't validate Path at all.
--->
-
-#### Rejected alternatives
-
-##### Portable regex behavior
-
-The safest route for specifying the regex would be to state a limited
-subset that can be used in a portable way. Any expressions outside of
-the subset will have implementation specific behavior.
-
-Regular expression subset (derived from [re2][re2-syntax] syntax page)
-
-| Expression | description             |
-|------------|-------------------------|
-| `.`        | any character           |
-| `[xyz]`    | character class         |
-| `[^xyz]`   | negated character class |
-| `x*`       | 0 or more x's           |
-| `x+`       | 1 or more x's           |
-| `xy`       | x followed by y         |
-| `x|y`      | x or y (prefer x)       |
-| `(abc)`    | grouping                |
+Interoperability between versions will be as follows:
 
-Maintaining a regular expression subset is likely not worth the complexity and
-is likely impossible across the [many implementations][regex-survey].
+1. v1beta1 will default to `ImplementationSpecific`.
+1. v1 will default to `Prefix`.
 
 ### API field naming
 
 These are straightforwarding one-to-one renames to for consistency.
 
-| v1beta1 field  | ga field              | rationale                   |
-|----------------|-----------------------|-----------------------------|
-| `spec.backend` | `spec.defaultBackend` | Explicitly mentions default |
+| v1beta1        | v1                    | Rationale                  |
+|----------------|-----------------------|----------------------------|
+| `spec.backend` | `spec.defaultBackend` | Explicitly mention default |
```

Comment:
How do we rename this safely? IIRC we may need to create a new field at this point in the API process.

Reply:
fields can be renamed across version boundaries by conversion